### ISSUE TITLE
How to work with circual type annotations?

### ISSUE THREAD

**Issue by Apollo3zehn** [microsoft/pylance-release#2419]
I am seeing below error for a simple sample with circular type hints.

![image](https://user-images.githubusercontent.com/20972129/155719339-fc9f067a-cf70-4083-aea4-52ca1b7ae421.png)

Is there a way to solve this? I tried `from __future__ import annotations` but that did not help. Additionally, adding `# pyright: reportUndefinedVariable=false` to the file suppresses the warning but that is not really what I want because then I loose that check for actually missing variables.

**Comment by erictraut** [comment]
[PEP 484](https://www.python.org/dev/peps/pep-0484/#forward-references) specifies that you can enclose a type annotation in quotes to indicate that it's a forward reference. Without doing so, your code will crash at runtime because the interpreter will attempt to evaluate `B`, and it will be undefined at that point. By enclosing it in quotes, the interpreter will interpret it as a string literal, but tools like static type checkers know that it should not be treated as a string.

```python
class A:
    b: "B"
```

**Comment by Apollo3zehn** [comment]
Thank you! The problem disappeared when I tried again to use the `from __future__ import annotations`. Maybe I had to restart VSCode or recompile. Somehow it is working now without quoting the types.

The thing is that I am working on a python code generator and there I do not know so easily which types depend on each other, so the future import is the easier solution for me. 

I will observe the behaviour, hopefully it keeps working.

Thanks again, Apollo3zehn
